### ISSUE TITLE
test_init and test_tpm2_init fail in a systemd-nspawn container when seccomp support is enabled

### ISSUE THREAD

**Issue by thithib** [stefanberger/swtpm#166]
Hello,

I maintain an [swtpm PKGBUILD on AUR](https://aur.archlinux.org/packages/swtpm/). I tried bumping to v0.2.0 and two tests are failing if seccomp support is enabled:

```
FAIL: test_init
===============

Need to be root to run test with CUSE interface.
Need to be root to run test with CUSE interface.
==== Starting swtpm with interfaces socket+socket ====
swtpm: SWTPM_NVRAM_LoadData: Error (fatal) opening /tmp/tmp.oyAiS5aVaZ/tpm-00.permall for read, Permission denied
swtpm: Error: Could not initialize libtpms.
swtpm: Error: Could not initialize the TPM
TPM result from PTM_INIT: 0x9
Process 8997 has wrong seccomp profile type
Expected: 0
Actual  : 2
_test_init: line 20:  8997 Killed                  ${SWTPM_EXE} socket $@ --server type=tcp,port=${SWTPM_SERVER_PORT}${swtpm_server_disconnect} --ctrl type=tcp,port=${SWTPM_CTRL_PORT}
FAIL test_init (exit status: 1)

FAIL: test_tpm2_init
====================

Need to be root to run test with CUSE interface.
Need to be root to run test with CUSE interface.
==== Starting swtpm with interfaces socket+socket ====
swtpm: SWTPM_NVRAM_LoadData: Error (fatal) opening /tmp/tmp.ui40lUuyb0/tpm2-00.permall for read, Permission denied
swtpm: SWTPM_NVRAM_LoadData: Error (fatal) opening /tmp/tmp.ui40lUuyb0/tpm2-00.permall for read, Permission denied
libtpms/tpm2: Entering failure mode; code: 8, location: NvPowerOn line 126
swtpm: Error: Could not initialize libtpms.
swtpm: Error: Could not initialize the TPM
TPM result from PTM_INIT: 0x101
Process 12430 has wrong seccomp profile type
Expected: 0
Actual  : 2
_test_tpm2_init: line 19: 12430 Killed                  ${SWTPM_EXE} socket $@ --server type=tcp,port=${SWTPM_SERVER_PORT}${swtpm_server_disconnect} --ctrl type=tcp,port=${SWTPM_CTRL_PORT}
FAIL test_tpm2_init (exit status: 1)
```

If I disable seccomp support in swtpm, tests pass fine.

I'm still investigating this issue, but there's one thing I find strange with your `check_seccomp_profile` function, especially when you call it from the two failing tests. You are checking that the process's seccomp mode is `SECCOMP_MODE_DISABLED`. Wouldn't that be wrong if the build environment was already using `SECCOMP_MODE_STRICT` or `SECCOMP_MODE_FILTER`?  Because I'm doing builds in a systemd-nspawn container and systemd-nspawn does launch containers with a default seccomp filter. When I build directly on my host system outside a container, tests are passing fine.

**Comment by stefanberger** [comment]
The test cases do not anticipate being start with a seccomp profile setup by the environment, like in your case. `_test_tpm2_init` sets the `--seccomp action=none` to test that no seccomp profile is activated, so this is intentional, though the test is failing now because there is a seccomp profile even though there should be none. The other failures above are testing that errors propagate as expected, so ignore them.

I guess we would have to check whether the bash script itself is running with a seccomp profile and handle this case accordingly in. Can you test this PR here: https://github.com/stefanberger/swtpm/pull/167

Patch: https://github.com/stefanberger/swtpm/commit/837f88946c438ceec1e3cfd9534af1cc190005a1.patch

**Comment by thithib** [comment]
> I guess we would have to check whether the bash script itself is running with a seccomp profile and handle this case accordingly in. Can you test this PR here: #167
> 
> Patch: https://github.com/stefanberger/swtpm/commit/837f88946c438ceec1e3cfd9534af1cc190005a1.patch

Yes, this patch fixes the issue. No more tests are failing, thanks!

**Comment by stefanberger** [comment]
It will be available in the stable-0.2.0 branch soon. Thanks for reporting it.

**Comment by stefanberger** [comment]
If your problem is solved, can you close this issue, please?
